### PR TITLE
AB Agent Improvements.

### PR DESCRIPTION
**Pull request by Lucaweihs**
(1) Bug fixes to limits code, (2) adding new interactive debug commands, (3) respecting the maxForce parameter when moving the arm and body, (4) resetting drive targets after moving.

### CODE REVIEW

File: unity/Assets/Scripts/ArticulatedAgentSolver.cs [hunk 1]
```diff
@@ -129,7 +133,13 @@ public class ArticulatedAgentSolver : MonoBehaviour, MovableContinuous {
                 float relativeForce = (speedDelta / Time.fixedDeltaTime) * currentAgentMoveParams.agentMass
```
Comment (AlvaroHG) [inline unity/Assets/Scripts/ArticulatedAgentSolver.cs:133]: Should be `fixedDeltaTime` one passed from the action

File: unity/Assets/Scripts/ArticulatedAgentSolver.cs [hunk 2]
```diff
@@ -144,8 +154,12 @@ public class ArticulatedAgentSolver : MonoBehaviour, MovableContinuous {
                 // if (relativeForce.sqrMagnitude > Mathf.Abs(maxForce.sqrMagnitude)) {
```
Comment (AlvaroHG) [inline unity/Assets/Scripts/ArticulatedAgentSolver.cs:154]: Same for above `Time.fixedDeltaTime` -> `fixedDeltaTime`

File: unity/Assets/Scripts/ArticulatedArmJointSolver.cs [hunk 3]
```diff
@@ -161,9 +162,9 @@ public class ArticulatedArmJointSolver : MonoBehaviour {
                     float distanceRemaining = currentArmMoveParams.distance - distanceMovedSoFar;
 
                     // New version of up-down drive
-                    float forceAppliedFromRest = 500f;
+                    float forceAppliedFromRest = currentArmMoveParams.maxForce;
                     float slowDownTime = 5 * fixedDeltaTime;
```
Comment (AlvaroHG) [inline unity/Assets/Scripts/ArticulatedArmJointSolver.cs:166]: Why do we have this `5` again?

File: unity/Assets/Scripts/ArticulatedArmJointSolver.cs [hunk 4]
```diff
@@ -266,14 +268,14 @@ public class ArticulatedArmJointSolver : MonoBehaviour {
...
-                    drive.damping = forceAppliedFromRest / currentArmMoveParams.speed;
+                    drive.damping = Mathf.Min(forceAppliedFromRest / currentArmMoveParams.speed, 10000f);
 
                     float signedDistanceRemaining = direction * distanceRemaining;
 
                     float maxSpeed = Mathf.Max(distanceRemaining / fixedDeltaTime, 0f); // Never move so fast we'll overshoot in 1 step
                     currentArmMoveParams.speed = Mathf.Min(maxSpeed, currentArmMoveParams.speed);
 
-                    float curVelocity = myAB.velocity.y;
+                    float curVelocity = myAB.velocity.z;
```
Comment (AlvaroHG) [inline unity/Assets/Scripts/ArticulatedArmJointSolver.cs:278]: Aha!

File: unity/Assets/Scripts/ArticulatedAgentSolver.cs [hunk 5]
```diff
@@ -129,7 +133,13 @@ public class ArticulatedAgentSolver : MonoBehaviour, MovableContinuous {
                 float relativeForce = (speedDelta / Time.fixedDeltaTime) * currentAgentMoveParams.agentMass
                     * currentAgentMoveParams.direction;
 
-                Debug.Log("3. speedDelta is " + speedDelta + ". Applying force of " + relativeForce);
+                relativeForce = Mathf.Clamp(relativeForce, -currentAgentMoveParams.maxForce, currentAgentMoveParams.maxForce);
+                Debug.Log(
+                    $"3. speedDelta is {speedDelta}. Applying force of {relativeForce} = "
+                    + $"clamp({currentAgentMoveParams.maxForce}, {speedDelta} / {Time.fixedDeltaTime}) * {currentAgentMoveParams.agentMass} * {currentAgentMoveParams.direction})."
```
Comment (AlvaroHG) [inline unity/Assets/Scripts/ArticulatedAgentSolver.cs:139]: Jus for debug but use `fixedDeltaTime` instead

File: unity/Assets/Scripts/ArticulatedAgentSolver.cs [hunk 6]
```diff
@@ -144,8 +154,12 @@ public class ArticulatedAgentSolver : MonoBehaviour, MovableContinuous {
                 // if (relativeForce.sqrMagnitude > Mathf.Abs(maxForce.sqrMagnitude)) {
                 //     relativeForce = maxForce;
                 // }
+                relativeForce = Mathf.Clamp(relativeForce, -currentAgentMoveParams.maxForce, currentAgentMoveParams.maxForce);
+                Debug.Log(
+                    $"2. speedDelta is {speedDelta}. Applying force of {relativeForce} = "
+                    + $"clamp({currentAgentMoveParams.maxForce}, {speedDelta} / {Time.fixedDeltaTime}) * {currentAgentMoveParams.agentMass} * {currentAgentMoveParams.direction})."
```
Comment (AlvaroHG) [inline unity/Assets/Scripts/ArticulatedAgentSolver.cs:160]: Same comment as above  `Time.fixedDeltaTime` ->`fixedDeltatTime`

**Comment by AlvaroHG** [comment]
Minor comments on surrounding code, LGTM

File: unity/Assets/Scripts/ArticulatedAgentSolver.cs (hunk 1 above)
Comment (Lucaweihs) [reply unity/Assets/Scripts/ArticulatedAgentSolver.cs]: Done

File: unity/Assets/Scripts/ArticulatedAgentSolver.cs (hunk 2 above)
Comment (Lucaweihs) [reply unity/Assets/Scripts/ArticulatedAgentSolver.cs:154]: Done

File: unity/Assets/Scripts/ArticulatedArmJointSolver.cs (hunk 3 above)
Comment (winthos) [reply unity/Assets/Scripts/ArticulatedArmJointSolver.cs:166]: morale support!

Comment (Lucaweihs) [reply unity/Assets/Scripts/ArticulatedArmJointSolver.cs:166]: Arbitrary, idea was to give the controller 5 updates to try to slow things down.

File: unity/Assets/Scripts/ArticulatedAgentSolver.cs (hunk 5 above)
Comment (Lucaweihs) [reply unity/Assets/Scripts/ArticulatedAgentSolver.cs]: Done

File: unity/Assets/Scripts/ArticulatedAgentSolver.cs (hunk 6 above)
Comment (Lucaweihs) [reply unity/Assets/Scripts/ArticulatedAgentSolver.cs]: Done

**Pull request merged**